sros2/sros2/api/__init__.py

```diff
@@ -364,16 +381,26 @@ def create_key(keystore_path, identity):
     create_permission_file(permissions_path, domain_id, policy_element)
 
     signed_permissions_path = os.path.join(key_dir, 'permissions.p7s')
-    keystore_permissions_ca_key_path = os.path.join(keystore_path, KS_PRIVATE, 'permissions_ca.key.pem')
+    keystore_permissions_ca_key_path = os.path.join(
+        keystore_path, KS_PRIVATE, 'permissions_ca.key.pem')
     _create_smime_signed_file(
-        keystore_ca_cert_path, keystore_permissions_ca_key_path, permissions_path, signed_permissions_path)
+        keystore_ca_cert_path,
+        keystore_permissions_ca_key_path,
+        permissions_path,
+        signed_permissions_path
+    )
 
     return True
 
 
 def list_keys(keystore_path):
-    for name in os.listdir(keystore_path):
-        if os.path.isdir(os.path.join(keystore_path, name)):
+    contexts_path = os.path.join(keystore_path, KS_CONTEXT)
+    if not os.path.exists(keystore_path):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), keystore_path)
+    if not os.path.exists(contexts_path):
+        return True
+    for name in os.listdir(contexts_path):
```

Comment:
Context paths can be nested. Is listdir recursive?
I.e. will it walk over contexts that aren't just in flat in the root of the context sub folder?

Reply:
> To be fair it was already the case before this change so doesn't have to be addressed here. Should be ticketed though if not addressed

Agreed.
What should be the output of:

```
-> contexts
  -> context1
    -> security artifacts ...
    -> nested1
      -> security artifacts ...
  ->context2
    ->nested2
      ->security artifacts...
```
?

IMO, it should be:

```
/context1
/context1/nested1
/context2/nested2
```

If the `contexts` folder have valid security artifacts, `/` should be showed too.

Reply:
Sounds good to me